include/parser.h

```diff
@@ -0,0 +1,25 @@
+#ifndef INCLUDE_PARSER_H_
+#define INCLUDE_PARSER_H_
```

Comment:
The variable naming scheme here is not standardized across source files. Please use `_FULLTEXT_NAME_H_` across the board (that is, this would be `_FULLTEXT_PARSER_H_`).